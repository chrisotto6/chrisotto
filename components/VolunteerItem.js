import kebabCase from '@/lib/utils/kebabCase'

const VolunteerItem = ({ company, title, summary, yearStart, yearEnd }) => {
  return (
    <div data-testid={`vol-${kebabCase(company)}`}>
      <h3>{company}</h3>
      <h4>{title}</h4>
      <h5>{`${yearStart} - ${yearEnd}`}</h5>
      <p>{summary}</p>
    </div>
  )
}

export default VolunteerItem
