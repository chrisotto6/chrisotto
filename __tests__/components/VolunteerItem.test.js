import { render } from '@testing-library/react'
import VolunteerItem from '@/components/VolunteerItem'

it('renders VolunteerItem correctly', () => {
  const vol = {
    company: 'Microsoft Teals',
    title: 'Volunteer Teacher',
    summary: 'test',
    yearStart: '2020',
    yearEnd: '2020',
  }
  const { container } = render(
    <VolunteerItem
      company={vol.company}
      title={vol.title}
      summary={vol.summary}
      yearStart={vol.yearStart}
      yearEnd={vol.yearEnd}
    />
  )
  expect(container).toMatchSnapshot()
})
