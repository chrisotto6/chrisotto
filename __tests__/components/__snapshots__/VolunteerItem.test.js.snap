// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders VolunteerItem correctly 1`] = `
<div>
  <div
    data-testid="vol-microsoft-teals"
  >
    <h3>
      Microsoft Teals
    </h3>
    <h4>
      Volunteer Teacher
    </h4>
    <h5>
      2020 - 2020
    </h5>
    <p>
      test
    </p>
  </div>
</div>
`;
